[ISSUE #9396] Use fastjson2 in all modules

yx9o opened the pull request:
Fixes #9396 .

fuyou001 commented:
1、Could you provide the performance benchmark data for the original Fastjson 1?
2、Could you  provide compatibility test results with Fastjson 1, particularly when the object contains camel-case properties?

yx9o commented:
> 1、Could you provide the performance benchmark data for the original Fastjson 1? 2、Could you provide compatibility test results with Fastjson 1, particularly when the object contains camel-case properties?

Thank you for your review. The data for the two questions are as follows:
1. 
<img width="1633" alt="image" src="https://github.com/user-attachments/assets/23fb4fa0-c383-4a41-9a14-a5337d1b7726" />

2. 
<img width="852" alt="image" src="https://github.com/user-attachments/assets/19109938-7a1b-4b3c-bbdf-00ed199a86f2" />
<img width="938" alt="image" src="https://github.com/user-attachments/assets/da117e3b-a15a-4c1c-ae31-8b7aeda2b589" />
<img width="676" alt="image" src="https://github.com/user-attachments/assets/34a63e07-a9b0-4659-b95f-1811042e0048" />

Regarding question 2, I encountered the camel case problem of getCId during the replacement test. The solution is to move the @JSONField(name = "c") annotation from the attribute to the get/set method. All the replaced codes are covered by unit tests.

fuyou001 commented:
Could you develop a tool to validate whether objects with camelCase properties are compatible with Fastjson 2, so we can check compatibility? @yx9o

yx9o commented:
> Could you develop a tool to validate whether objects with camelCase properties are compatible with Fastjson 2, so we can check compatibility? @yx9o

Let me understand. We need to develop a tool to test all objects in the project that have irregular camel case naming to ensure that fastjson1 and fastjson2 serialization are compatible, right?

fuyou001 commented:
> > Could you develop a tool to validate whether objects with camelCase properties are compatible with Fastjson 2, so we can check compatibility? @yx9o
> 
> Let me understand. We need to develop a tool to test all objects in the project that have irregular camel case naming to ensure that fastjson1 and fastjson2 serialization are compatible, right?

Yes, but we should only validate classes involved in JSON serialization/deserialization.

yx9o commented:
Hi @fuyou001 , please help review again. According to your review, I added the following serialization compatibility test tool class: `RemotingSerializableCompatTest`. It mainly does the following:
1. Construct objects with default parameters for all subclasses of RemotingSerializable
2. Use fastjson1 to serialize the object obtained in the first step
3. Use fastjson2 to deserialize the json string obtained in the second step
4. Compare the object deserialized in the third step with the object constructed in the first step, and compare all attribute values. Their comparisons are all passed

fuyou001 commented:
> Hi @fuyou001 , please help review again. According to your review, I added the following serialization compatibility test tool class: `RemotingSerializableCompatTest`. It mainly does the following:
> 
> 1. Construct objects with default parameters for all subclasses of RemotingSerializable
> 2. Use fastjson1 to serialize the object obtained in the first step
> 3. Use fastjson2 to deserialize the json string obtained in the second step
> 4. Compare the object deserialized in the third step with the object constructed in the first step, and compare all attribute values. Their comparisons are all passed

When running RemotingSerializableCompatTest, I encountered failed classes. Here's the output for your review:
failed:class org.apache.rocketmq.remoting.protocol.body.RegisterBrokerBody
failed:class org.apache.rocketmq.remoting.protocol.header.GetConsumerListByGroupResponseBody
failed:class org.apache.rocketmq.remoting.protocol.route.TopicRouteData
failed:class org.apache.rocketmq.remoting.protocol.statictopic.TopicRemappingDetailWrapper
failed:class org.apache.rocketmq.remoting.protocol.body.SubscriptionGroupList
failed:class org.apache.rocketmq.remoting.protocol.body.GroupList
failed:class org.apache.rocketmq.remoting.protocol.body.ElectMasterResponseBody
failed:class org.apache.rocketmq.remoting.protocol.body.QueryAssignmentResponseBody
failed:class org.apache.rocketmq.remoting.protocol.body.BrokerMemberGroup
failed:class org.apache.rocketmq.remoting.protocol.heartbeat.HeartbeatData
failed:class org.apache.rocketmq.remoting.protocol.body.CheckClientRequestBody
failed:class org.apache.rocketmq.remoting.protocol.body.GetBrokerMemberGroupResponseBody
failed:class org.apache.rocketmq.remoting.protocol.body.ClusterInfo
failed:class org.apache.rocketmq.remoting.protocol.body.QueryConsumeQueueResponseBody
failed:class org.apache.rocketmq.remoting.protocol.body.TopicList
failed:class org.apache.rocketmq.remoting.protocol.body.QuerySubscriptionResponseBody
failed:class org.apache.rocketmq.remoting.protocol.body.ConsumeByWho
failed:class org.apache.rocketmq.remoting.protocol.body.ConsumerRunningInfo
failed:class org.apache.rocketmq.remoting.protocol.body.KVTable
failed:class org.apache.rocketmq.remoting.protocol.body.CreateTopicListRequestBody

fuyou001 commented:
@yx9o  Subclasses of ConfigManager must also be checked

yx9o commented:
> > Hi @fuyou001 , please help review again. According to your review, I added the following serialization compatibility test tool class: `RemotingSerializableCompatTest`. It mainly does the following:
> > 
> > 1. Construct objects with default parameters for all subclasses of RemotingSerializable
> > 2. Use fastjson1 to serialize the object obtained in the first step
> > 3. Use fastjson2 to deserialize the json string obtained in the second step
> > 4. Compare the object deserialized in the third step with the object constructed in the first step, and compare all attribute values. Their comparisons are all passed
> 
> When running RemotingSerializableCompatTest, I encountered failed classes. Here's the output for your review: failed:class org.apache.rocketmq.remoting.protocol.body.RegisterBrokerBody failed:class org.apache.rocketmq.remoting.protocol.header.GetConsumerListByGroupResponseBody failed:class org.apache.rocketmq.remoting.protocol.route.TopicRouteData failed:class org.apache.rocketmq.remoting.protocol.statictopic.TopicRemappingDetailWrapper failed:class org.apache.rocketmq.remoting.protocol.body.SubscriptionGroupList failed:class org.apache.rocketmq.remoting.protocol.body.GroupList failed:class org.apache.rocketmq.remoting.protocol.body.ElectMasterResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.QueryAssignmentResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.BrokerMemberGroup failed:class org.apache.rocketmq.remoting.protocol.heartbeat.HeartbeatData failed:class org.apache.rocketmq.remoting.protocol.body.CheckClientRequestBody failed:class org.apache.rocketmq.remoting.protocol.body.GetBrokerMemberGroupResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.ClusterInfo failed:class org.apache.rocketmq.remoting.protocol.body.QueryConsumeQueueResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.TopicList failed:class org.apache.rocketmq.remoting.protocol.body.QuerySubscriptionResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.ConsumeByWho failed:class org.apache.rocketmq.remoting.protocol.body.ConsumerRunningInfo failed:class org.apache.rocketmq.remoting.protocol.body.KVTable failed:class org.apache.rocketmq.remoting.protocol.body.CreateTopicListRequestBody

Hello, I received your response. I have tested this class separately in the current branch and the develop branch. No errors occur. There is no "failed: xx" error output in the class itself. May I ask how you run this test?

yx9o commented:
> @yx9o Subclasses of ConfigManager must also be checked

ConfigManager's subclasses are in multiple different packages and cannot be tested uniformly using tool classes. I want to do fastjson1 serialization and fastjson2 deserialization compatibility verification in each subclass's Test. Is this OK?

fuyou001 commented:
> > > Hi @fuyou001 , please help review again. According to your review, I added the following serialization compatibility test tool class: `RemotingSerializableCompatTest`. It mainly does the following:
> > > 
> > > 1. Construct objects with default parameters for all subclasses of RemotingSerializable
> > > 2. Use fastjson1 to serialize the object obtained in the first step
> > > 3. Use fastjson2 to deserialize the json string obtained in the second step
> > > 4. Compare the object deserialized in the third step with the object constructed in the first step, and compare all attribute values. Their comparisons are all passed
> > 
> > 
> > When running RemotingSerializableCompatTest, I encountered failed classes. Here's the output for your review: failed:class org.apache.rocketmq.remoting.protocol.body.RegisterBrokerBody failed:class org.apache.rocketmq.remoting.protocol.header.GetConsumerListByGroupResponseBody failed:class org.apache.rocketmq.remoting.protocol.route.TopicRouteData failed:class org.apache.rocketmq.remoting.protocol.statictopic.TopicRemappingDetailWrapper failed:class org.apache.rocketmq.remoting.protocol.body.SubscriptionGroupList failed:class org.apache.rocketmq.remoting.protocol.body.GroupList failed:class org.apache.rocketmq.remoting.protocol.body.ElectMasterResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.QueryAssignmentResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.BrokerMemberGroup failed:class org.apache.rocketmq.remoting.protocol.heartbeat.HeartbeatData failed:class org.apache.rocketmq.remoting.protocol.body.CheckClientRequestBody failed:class org.apache.rocketmq.remoting.protocol.body.GetBrokerMemberGroupResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.ClusterInfo failed:class org.apache.rocketmq.remoting.protocol.body.QueryConsumeQueueResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.TopicList failed:class org.apache.rocketmq.remoting.protocol.body.QuerySubscriptionResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.ConsumeByWho failed:class org.apache.rocketmq.remoting.protocol.body.ConsumerRunningInfo failed:class org.apache.rocketmq.remoting.protocol.body.KVTable failed:class org.apache.rocketmq.remoting.protocol.body.CreateTopicListRequestBody
> 
> Hello, I received your response. I have tested this class separately in the current branch and the develop branch. No errors occur. There is no "failed: xx" error output in the class itself. May I ask how you run this test?

RemotingSerializableCompatTest.testCompatibilityCheck  method , I change :

![image](https://github.com/user-attachments/assets/f0905275-23f0-4f50-9a7e-751009c4ec62)

                boolean isSuccess = checkCompatible(instance, clazz);
                if (!isSuccess) {
                    System.out.println("failed:" + clazz);
                }

fuyou001 commented:
> > > > > Hi @fuyou001 , please help review again. According to your review, I added the following serialization compatibility test tool class: `RemotingSerializableCompatTest`. It mainly does the following:
> > > > > 
> > > > > 1. Construct objects with default parameters for all subclasses of RemotingSerializable
> > > > > 2. Use fastjson1 to serialize the object obtained in the first step
> > > > > 3. Use fastjson2 to deserialize the json string obtained in the second step
> > > > > 4. Compare the object deserialized in the third step with the object constructed in the first step, and compare all attribute values. Their comparisons are all passed
> > > > 
> > > > 
> > > > When running RemotingSerializableCompatTest, I encountered failed classes. Here's the output for your review: failed:class org.apache.rocketmq.remoting.protocol.body.RegisterBrokerBody failed:class org.apache.rocketmq.remoting.protocol.header.GetConsumerListByGroupResponseBody failed:class org.apache.rocketmq.remoting.protocol.route.TopicRouteData failed:class org.apache.rocketmq.remoting.protocol.statictopic.TopicRemappingDetailWrapper failed:class org.apache.rocketmq.remoting.protocol.body.SubscriptionGroupList failed:class org.apache.rocketmq.remoting.protocol.body.GroupList failed:class org.apache.rocketmq.remoting.protocol.body.ElectMasterResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.QueryAssignmentResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.BrokerMemberGroup failed:class org.apache.rocketmq.remoting.protocol.heartbeat.HeartbeatData failed:class org.apache.rocketmq.remoting.protocol.body.CheckClientRequestBody failed:class org.apache.rocketmq.remoting.protocol.body.GetBrokerMemberGroupResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.ClusterInfo failed:class org.apache.rocketmq.remoting.protocol.body.QueryConsumeQueueResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.TopicList failed:class org.apache.rocketmq.remoting.protocol.body.QuerySubscriptionResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.ConsumeByWho failed:class org.apache.rocketmq.remoting.protocol.body.ConsumerRunningInfo failed:class org.apache.rocketmq.remoting.protocol.body.KVTable failed:class org.apache.rocketmq.remoting.protocol.body.CreateTopicListRequestBody
> > > 
> > > 
> > > Hello, I received your response. I have tested this class separately in the current branch and the develop branch. No errors occur. There is no "failed: xx" error output in the class itself. May I ask how you run this test?
> > 
> > 
> > RemotingSerializableCompatTest.testCompatibilityCheck method , I change :
> > ![image](https://private-user-images.githubusercontent.com/685537/457755008-f0905275-23f0-4f50-9a7e-751009c4ec62.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA3MjYzNzIsIm5iZiI6MTc1MDcyNjA3MiwicGF0aCI6Ii82ODU1MzcvNDU3NzU1MDA4LWYwOTA1Mjc1LTIzZjAtNGY1MC05YTdlLTc1MTAwOWM0ZWM2Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYyNFQwMDQ3NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYzVmOGJiZGYyOGRlYTNhMDg4ZDQxOWY1OGYzMWMxMmFkZmJjZGViZDFlNDdmMGU0ODIxNmM3OGFjZGI4NDIzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.X2sRH-o3sPR6ZG4wWfoFK84H1Z5JFvqU6dGdfAuVnSs)
> > ```
> >             boolean isSuccess = checkCompatible(instance, clazz);
> >             if (!isSuccess) {
> >                 System.out.println("failed:" + clazz);
> >             } 
> > ```
> 
> I changed it like this and there was no error. Is there a difference in the environment? May I ask what branch, platform and jdk you are running on?


This PR branch and uses JDK 11.


There are many issues concerning compatibility reported at https://github.com/alibaba/fastjson2/issues. 


Be extremely cautiousUpgrading to it involves high risks

yx9o commented:
> > > > > > Hi @fuyou001 , please help review again. According to your review, I added the following serialization compatibility test tool class: `RemotingSerializableCompatTest`. It mainly does the following:
> > > > > > 
> > > > > > 1. Construct objects with default parameters for all subclasses of RemotingSerializable
> > > > > > 2. Use fastjson1 to serialize the object obtained in the first step
> > > > > > 3. Use fastjson2 to deserialize the json string obtained in the second step
> > > > > > 4. Compare the object deserialized in the third step with the object constructed in the first step, and compare all attribute values. Their comparisons are all passed
> > > > > 
> > > > > 
> > > > > When running RemotingSerializableCompatTest, I encountered failed classes. Here's the output for your review: failed:class org.apache.rocketmq.remoting.protocol.body.RegisterBrokerBody failed:class org.apache.rocketmq.remoting.protocol.header.GetConsumerListByGroupResponseBody failed:class org.apache.rocketmq.remoting.protocol.route.TopicRouteData failed:class org.apache.rocketmq.remoting.protocol.statictopic.TopicRemappingDetailWrapper failed:class org.apache.rocketmq.remoting.protocol.body.SubscriptionGroupList failed:class org.apache.rocketmq.remoting.protocol.body.GroupList failed:class org.apache.rocketmq.remoting.protocol.body.ElectMasterResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.QueryAssignmentResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.BrokerMemberGroup failed:class org.apache.rocketmq.remoting.protocol.heartbeat.HeartbeatData failed:class org.apache.rocketmq.remoting.protocol.body.CheckClientRequestBody failed:class org.apache.rocketmq.remoting.protocol.body.GetBrokerMemberGroupResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.ClusterInfo failed:class org.apache.rocketmq.remoting.protocol.body.QueryConsumeQueueResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.TopicList failed:class org.apache.rocketmq.remoting.protocol.body.QuerySubscriptionResponseBody failed:class org.apache.rocketmq.remoting.protocol.body.ConsumeByWho failed:class org.apache.rocketmq.remoting.protocol.body.ConsumerRunningInfo failed:class org.apache.rocketmq.remoting.protocol.body.KVTable failed:class org.apache.rocketmq.remoting.protocol.body.CreateTopicListRequestBody
> > > > 
> > > > 
> > > > Hello, I received your response. I have tested this class separately in the current branch and the develop branch. No errors occur. There is no "failed: xx" error output in the class itself. May I ask how you run this test?
> > > 
> > > 
> > > RemotingSerializableCompatTest.testCompatibilityCheck method , I change :
> > > ![image](https://private-user-images.githubusercontent.com/685537/457755008-f0905275-23f0-4f50-9a7e-751009c4ec62.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA3MjYzNzIsIm5iZiI6MTc1MDcyNjA3MiwicGF0aCI6Ii82ODU1MzcvNDU3NzU1MDA4LWYwOTA1Mjc1LTIzZjAtNGY1MC05YTdlLTc1MTAwOWM0ZWM2Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYyNFQwMDQ3NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYzVmOGJiZGYyOGRlYTNhMDg4ZDQxOWY1OGYzMWMxMmFkZmJjZGViZDFlNDdmMGU0ODIxNmM3OGFjZGI4NDIzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.X2sRH-o3sPR6ZG4wWfoFK84H1Z5JFvqU6dGdfAuVnSs)
> > > ```
> > >             boolean isSuccess = checkCompatible(instance, clazz);
> > >             if (!isSuccess) {
> > >                 System.out.println("failed:" + clazz);
> > >             } 
> > > ```
> > 
> > 
> > I changed it like this and there was no error. Is there a difference in the environment? May I ask what branch, platform and jdk you are running on?
> 
> This PR branch and uses JDK 11.
> 
> There are many issues concerning compatibility reported at https://github.com/alibaba/fastjson2/issues.
> 
> Be extremely cautious Upgrading to fastjson2 involves high risks

I have passed the debugging under jdk11, please help review it again, thank you.

fuyou001 commented:
@yx9o The conflicts need to be handled
<img width="1800" height="596" alt="image" src="https://github.com/user-attachments/assets/4e2aa689-115a-4116-a174-5df101f53eb8" />

yx9o commented:
> @yx9o The conflicts need to be handled <img alt="image" width="1800" height="596" src="https://private-user-images.githubusercontent.com/685537/468441336-4e2aa689-115a-4116-a174-5df101f53eb8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMwNzMwMzYsIm5iZiI6MTc1MzA3MjczNiwicGF0aCI6Ii82ODU1MzcvNDY4NDQxMzM2LTRlMmFhNjg5LTExNWEtNDExNi1hMTc0LTVkZjEwMWY1M2ViOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyMVQwNDM4NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NTgwM2U2NjNkNmUyNzNjOGVlYWIzZjk5YWZhZGM1YzYxNzFhOTdhMDhkYzY5NDQ1Y2UzYmZiYTRiMGQyYjgzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2UK6p07K0blXKPc2gkxnu-P55_NpMwc1wRXvZra_jfE">

The conflict has been resolved, please review again, thank you.

fuyou001 commented:
All fields in classes used for Fastjson serialization/deserialization must be annotated with @JSONField, 
requiring verification tools for annotation compliance.

@yx9o

guyinyou commented:
<img width="2626" height="294" alt="image" src="https://github.com/user-attachments/assets/80c0f232-2ee4-435f-b6e8-0d8610a877d4" />
<img width="2928" height="662" alt="image" src="https://github.com/user-attachments/assets/fd13bff2-02ba-4584-9b24-9e46fc697ff4" />
@yx9o 
Some problems were encountered during testing

yx9o commented:
> <img alt="image" width="2000" height="294" src="https://private-user-images.githubusercontent.com/36399867/487654172-80c0f232-2ee4-435f-b6e8-0d8610a877d4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgzMDA5NTgsIm5iZiI6MTc1ODMwMDY1OCwicGF0aCI6Ii8zNjM5OTg2Ny80ODc2NTQxNzItODBjMGYyMzItMmVlNC00MzVmLWI2ZTgtMGQ4NjEwYTg3N2Q0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTE5VDE2NTA1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFmNmM1Zjc0MDA0ZjFhMjZlYWQwMmUyMzFiZWRmNTgyMjFiNjkxNGQyNjdhZTViZjg0ZDc0MDc1YWM1ZTFjZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6iK9wLN1c5V2LwxsbCCczalIzoCX3w6T_lsxlfs6S7Q"> <img alt="image" width="2000" height="662" src="https://private-user-images.githubusercontent.com/36399867/487654234-fd13bff2-02ba-4584-9b24-9e46fc697ff4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgzMDA5NTgsIm5iZiI6MTc1ODMwMDY1OCwicGF0aCI6Ii8zNjM5OTg2Ny80ODc2NTQyMzQtZmQxM2JmZjItMDJiYS00NTg0LTliMjQtOWU0NmZjNjk3ZmY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTE5VDE2NTA1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5MmU1OTIwMzkzNWVjYTNiN2Y2YjIwODc3OGU2NTdkNTQxOTgxNzhmZmZmM2NjYTZjZjQxZDgxNTRkMWI0YjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.t_WMb3VEjFXNzKrbEHNwyrC1WByrLjKdhdCCqU2Na4M"> @yx9o Some problems were encountered during testing

Corrected, please review and verify again, thank you.

guyinyou commented:
@yx9o Some problems were encountered during testing yet

com.alibaba.fastjson2.JSONException: not support input EA==, offset 22, character ", line 1, column 23, fastjson-version 2.0.43 {"acks":[{"b":"EA==","c":"gid-72","it":60000,"pt":1758617828232,"q":0,"r":"0","rq":4,"so":245397,"t":"topic-a72"}],"brokerName":"rocketmq-broker-rmq-cn-htq4fg13h09-0"}
        at com.alibaba.fastjson2.JSONReader.readBinary(JSONReader.java:756)
        at com.alibaba.fastjson2.reader.ObjectReaderImplBitSet.readObject(ObjectReaderImplBitSet.java:44)
        at com.alibaba.fastjson2.reader.ObjectReaderImplBitSet.readObject(ObjectReaderImplBitSet.java:11)
        at com.alibaba.fastjson2.reader.ORG_20_18_BatchAck.readObject(Unknown Source)
        at com.alibaba.fastjson2.reader.ORG_18_2_BatchAckMessageRequestBody.readObject(Unknown Source)
        at com.alibaba.fastjson2.JSON.parseObject(JSON.java:1365)
        at org.apache.rocketmq.remoting.protocol.RemotingSerializable.fromJson(RemotingSerializable.java:64)
        at org.apache.rocketmq.remoting.protocol.RemotingSerializable.decode(RemotingSerializable.java:48)
        at org.apache.rocketmq.broker.processor.AckMessageProcessor.processRequest(AckMessageProcessor.java:170)
        at org.apache.rocketmq.broker.processor.AckMessageProcessor.processRequest(AckMessageProcessor.java:114)
        at org.apache.rocketmq.remoting.netty.NettyRemotingAbstract.lambda$buildProcessRequestHandler$2(NettyRemotingAbstract.java:426)
        at org.apache.rocketmq.remoting.netty.RequestTask.run(RequestTask.java:80)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:991)

yx9o commented:
> @yx9o Some problems were encountered during testing yet
> 
> com.alibaba.fastjson2.JSONException: not support input EA==, offset 22, character ", line 1, column 23, fastjson-version 2.0.43 {"acks":[{"b":"EA==","c":"gid-72","it":60000,"pt":1758617828232,"q":0,"r":"0","rq":4,"so":245397,"t":"topic-a72"}],"brokerName":"rocketmq-broker-rmq-cn-htq4fg13h09-0"} at com.alibaba.fastjson2.JSONReader.readBinary(JSONReader.java:756) at com.alibaba.fastjson2.reader.ObjectReaderImplBitSet.readObject(ObjectReaderImplBitSet.java:44) at com.alibaba.fastjson2.reader.ObjectReaderImplBitSet.readObject(ObjectReaderImplBitSet.java:11) at com.alibaba.fastjson2.reader.ORG_20_18_BatchAck.readObject(Unknown Source) at com.alibaba.fastjson2.reader.ORG_18_2_BatchAckMessageRequestBody.readObject(Unknown Source) at com.alibaba.fastjson2.JSON.parseObject(JSON.java:1365) at org.apache.rocketmq.remoting.protocol.RemotingSerializable.fromJson(RemotingSerializable.java:64) at org.apache.rocketmq.remoting.protocol.RemotingSerializable.decode(RemotingSerializable.java:48) at org.apache.rocketmq.broker.processor.AckMessageProcessor.processRequest(AckMessageProcessor.java:170) at org.apache.rocketmq.broker.processor.AckMessageProcessor.processRequest(AckMessageProcessor.java:114) at org.apache.rocketmq.remoting.netty.NettyRemotingAbstract.lambda$buildProcessRequestHandler$2(NettyRemotingAbstract.java:426) at org.apache.rocketmq.remoting.netty.RequestTask.run(RequestTask.java:80) at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) at java.base/java.lang.Thread.run(Thread.java:991)

The adjustment has been completed, please review again, thank you.

guyinyou commented:
希望增加bitset的兼容性测试，在目前的测试中该字段是null，可以增加模拟值去比较兼容性。
<img width="2154" height="806" alt="image" src="https://github.com/user-attachments/assets/196f039f-9a76-4cf5-b159-d540648e7491" />

yx9o commented:
> 希望增加bitset的兼容性测试，在目前的测试中该字段是null，可以增加模拟值去比较兼容性。 <img alt="image" width="2000" height="806" src="https://private-user-images.githubusercontent.com/36399867/502458080-196f039f-9a76-4cf5-b159-d540648e7491.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA3MTU2NzIsIm5iZiI6MTc2MDcxNTM3MiwicGF0aCI6Ii8zNjM5OTg2Ny81MDI0NTgwODAtMTk2ZjAzOWYtOWE3Ni00Y2Y1LWIxNTktZDU0MDY0OGU3NDkxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDE3VDE1MzYxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEwN2JjOWUzN2Y3ZTMwYzE3NTg0NmUxYzE1M2Y3NWUyYzA5NWRhNDk0NWVjZDBhYTAwMzVjNDNhNDJjZWE2MGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._YXb2sVsiLmHJ2nNKYytXE73dwNUIVVJGsh7-Yvz0Js">

Added, please review again, thank you.
